File: C#/inorder_traversal.cs
using System;

public class TreeNode
{
    public int val;
    public TreeNode left;
    public TreeNode right;
    public TreeNode(int x) { val = x; }
}

public class BinaryTreeInorderTraversal
{
    public void InorderTraversal(TreeNode root)
    {
        if (root != null)
        {
            InorderTraversal(root.left);
            Console.Write(root.val + " ");
            InorderTraversal(root.right);
        }
    }

    public static void Main()
    {
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);
        root.left.right = new TreeNode(5);

        BinaryTreeInorderTraversal traversal = new BinaryTreeInorderTraversal();
        Console.Write("Inorder traversal of binary tree: ");
        traversal.InorderTraversal(root);
        Console.WriteLine();
    }
}

File: Python/Inorder_traversal.py
class TreeNode:
    def __init__(self, value=0, left=None, right=None):
        self.value = value
        self.left = left
        self.right = right

def inorder_traversal(root):
    if root:
        inorder_traversal(root.left)
        print(root.value, end=" ")
        inorder_traversal(root.right)

if __name__ == "__main__":
    # Construct a binary tree
    root = TreeNode(1)
    root.right = TreeNode(2)
    root.right.left = TreeNode(3)
    
    print("Inorder traversal: ", end="")
    inorder_traversal(root)
    print()
